CONTRIBUTING.rst

```diff
@@ -0,0 +1,247 @@
+* *OPT* : optimization
+* *BRK* : breaks something and/or tests fail
+* *DOC*: for all kinds of documentation related commits
+* *TST* : for adding or changing tests
+* *DAT* : for adding or changing data files
+* *STY* : PEP8 conformance, whitespace changes etc that do not affect
+  function.
+
+So your commit message might look something like this::
+
+    TST: relax test threshold slightly
+
+    Attempted fix for failure on windows test run when arrays are in fact
+    very close (within 6 dp).
+
+Keeping up a habit of doing this is useful because it makes it much easier to
+see at a glance which changes are likely to be important when you are looking
+for sources of bugs, fixes, large refactorings or new features.
+
+Pull request codes
+------------------
+
+When you submit a pull request to github, github will ask you for a summary.  If
+your code is not ready to merge, but you want to get feedback, please consider
+using ``WIP - experimental optimization`` or similar for the title of your pull
+request. That way we will all know that it's not yet ready to merge and that
+you may be interested in more fundamental comments about design.
+
+When you think the pull request is ready to merge, change the title (using the
+*Edit* button) to something like ``MRG - optimization``.
+
+Test coverage
+-------------
+
+Tests for a module should ideally cover all code in that module,
+i.e., statement coverage should be at 100%.
+
+To measure the test coverage, install
+`coverage.py <http://nedbatchelder.com/code/coverage/>`__
+(e.g., using ``pip install coverage``) and then run::
+
+  $ make coverage
+
+This will print a report with one line for each file in `mousestyles`,
+detailing the test coverage::
+
+  Name                 Stmts   Miss Branch BrMiss  Cover   Missing
+  ----------------------------------------------------------------
+  mousestyles                 43      6     10      1    87%   72, 77-88
+  mousestyles.core            55      0     30      4    95%
+  mousestyles.data            45      0      2      0   100%
+  mousestyles.eda             22      0      8      0   100%
+  mousestyles.irr             52      0     20      2    97%
+  mousestyles.stratified      44      0     16      4    93%
+  ----------------------------------------------------------------
+  TOTAL                  261      6     86     11    95%
+  ----------------------------------------------------------------------
+  Ran 35 tests in 37.199s
+
+  OK
+
+Bugs
+----
+
+Please `report bugs on GitHub <https://github.com/berkeley-stat222/mousestyles/issues>`_.
+
+Working with ``np.random``
+--------------------------
+
+If you need to set a random seed, do **not** use ``np.random.seed()``.  Using
+``np.random.seed()`` can produce inconsistent results.  Instead you should use
+:func:`get_prng` with your chosen seed.  Here is a tiny
+example of how you might create a function that generates random numbers::
+
+    from mousestyles.utils import get_prng

```

Comment:
The 'get_prng' function is associated with the **permute** package. We can consider removing this final section after discussing with @jarrodmillman.